allennlp/data/dataset_readers/dataset_utils/text2sql_utils.py

```diff
@@ -78,6 +82,25 @@ def clean_and_split_sql(sql: str) -> List[str]:
             sql_tokens.extend(split_table_and_column_names(token))
     return sql_tokens
 
+def resolve_primary_keys_in_schema(sql_tokens: List[str],
+                                   schema: Dict[str, List[TableColumn]]) -> List[str]:
+    """
+    Some examples in the text2sql datasets use ID as a column reference to the
```

Comment:
What's wrong with just adding `ID` as another column that can be generated?

Reply:
Because it's essentially a floating reference to any column and I think it would be hard for a model to learn that. Also, I have a strong bias to as few actions as possible, and this ID is not really necessary.